### PR TITLE
Add mixed periodic boundary condition support to EwaldCalculator

### PR DESCRIPTION
**Pull request by E-Rum**
PR to introduce mixed PBC

# Contributor (creator of pull-request) checklist

 - [x] Tests updated (for new features and bugfixes)?
 - [x] Documentation updated (for new features)?
 - [x] Issue referenced (for PRs that solve an issue)?

# Reviewer checklist

 - [x] CHANGELOG updated with public API or any other important changes?

----
📚 Documentation preview 📚: https://torch-pme--200.org.readthedocs.build/en/200/

### CODE REVIEW

**Comment by E-Rum** [comment]
This is the first step. For now, the Yeh–Berkowitz correction has been implemented, supporting 2D structures by correcting the classical 3D Ewald method.

The next step is to look for tests that ensure the correctness of the implementation and also to write corrections for the 1D and 0D cases.

File: src/torchpme/calculators/pme.py [hunk 1]
```diff
@@ -97,51 +98,82 @@ def _compute_kspace(
         charges: torch.Tensor,
         cell: torch.Tensor,
         positions: torch.Tensor,
+        periodic: Optional[torch.Tensor] = None,
     ) -> torch.Tensor:
+        if periodic is None:
```
Comment (PicoCentauri) [inline src/torchpme/calculators/pme.py:103]: Should we maybe move this in a private function to avoid duplications?

File: src/torchpme/calculators/pme.py [hunk 2]
```diff
@@ -97,51 +98,82 @@ def _compute_kspace(
...
+        # If the cell has a net charge (i.e. if sum(charges) != 0), the method
+        # implicitly assumes that a homogeneous background charge of the opposite
+        # sign is present to make the cell neutral. In this case, the potential has
+        # to be adjusted to compensate for this. An extra factor of 2 is added to
+        # compensate for the division by 2 later on
+        charge_tot = torch.sum(charges, dim=0)
+        prefac = self.potential.background_correction()
+        interpolated_potential -= 2 * prefac * charge_tot * ivolume
+
+        if periodicity == 2:
```
Comment (PicoCentauri) [inline src/torchpme/calculators/pme.py:164]: This could also be a function.

File: tests/calculators/test_workflow.py [hunk 3]
```diff
@@ -183,6 +184,83 @@ def test_not_nan(self, CalculatorClass, params, device, dtype):
...
+
+    def test_periodicity_true_value(
+        self,
+        CalculatorClass,
+        params,
+        device,
+        dtype,
+    ):
+        """Test that values coincide with values from LAMMPS"""
+        true_value = -383.44635
```
Comment (PicoCentauri) [inline tests/calculators/test_workflow.py:238]: I don't want to overdo these tests but maybe we can do a better test as just a regression one.

I will think about it.

**Review by PicoCentauri — COMMENTED** [review]
Very nice. Some early comments. Maybe we can avoid some code duplications...
